### DISCUSSION TITLE
kong 3.0.0 started success, but access error

### DISCUSSION THREAD

**Discussion by IMMOLK** [Kong/kong#9548] [Help] [answered]
![image](https://user-images.githubusercontent.com/75161356/195592852-71e9d70c-53b7-485d-adf9-155a0d4c5472.png)

**Comment by hanshuebner** [comment]
Hello @IMMOLK,

how did you start Kong?  Can you share the configuration that you used and details about the setup?

Thank you,
Hans

**Reply by IMMOLK** [reply]
I installed kong with rpm package
kong version:  `3.0.0`
linux version: `Linux version 5.4.17-2102.201.3.el7uek.x86_64 (mockbuild@host-100-100-224-4) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-39.0.3) (GCC)) #2 SMP Fri Apr 23 09:05:55 PDT 2021
`

I used the defautl kong.conf, and there are my modifications
`database = off;`
`declarative_config = /home/xxx/kong.yml;`

I generated the yam file with the command `kong config init`

start commond :
`kong start -c /home/xxx/kong.conf -p /home/xxx/kong --nginx-conf /home/xxx/nginx.conf`

i access the url which contains "localhost:8000"  or "localhost:8001" failed, but  the url which was configure in nginx.conf i can access.

error message: `failed to create and open LMDB database: no LMDB environment defined;unable to open DB for access: no LMDB environment defined`

Do I need to install some additional content ?

![image](https://user-images.githubusercontent.com/75161356/195740056-c58b3588-31bc-43be-9f7f-b73b26959460.png)

**Reply by StarlightIbuki** [reply]
It looks like the LMDB is not correctly initialized. You should be able to find errors related to LMDB earlier than that one. The most likely reason would be that the file LMDB trying to write is not writable (permission, parent directory not existing, etc. Try to find `lmdb_environment_path` in your configuration).

**Reply by michaelhsk** [reply]
I got the same error
```
[crit] 350#0: *4 [lua] init.lua:678: init_worker(): worker initialization error: failed to create and open LMDB database: no LMDB environment defined; this node must be restarted, context: init_worker_by_lua*
```
after upgrade from 2.8.1 to 3.0.0 and try to start my dataplane.

My Poblem was that i used a modified template to generate nginx.conf. They added this lines to the original template nginx.lua

> if database == "off" then
lmdb_environment_path ${{LMDB_ENVIRONMENT_PATH}};
lmdb_map_size         ${{LMDB_MAP_SIZE}};
> end

After adding this lines to my modified template the error was fixed.

**Comment by slastique** [comment, marked as answer]
I've faced the same issue (upgrade from 2.8 to 3.3.0). my solution is to add the following lines into nginx.conf
```
    lmdb_environment_path dbless.lmdb;
    lmdb_map_size         128m;
```
right after these lines
```
pid pids/nginx.pid;
error_log /dev/stderr notice;

# injected nginx_main_* directives
```

i've compared files from kong:3.3.0-ubuntu image and my configuration and found this difference.

**Reply by IMMOLK** [reply]
thank you, it dose work
